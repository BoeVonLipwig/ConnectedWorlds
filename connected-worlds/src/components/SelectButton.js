import React from 'react';

class SelectButton extends React.Component {
  render() {
    return (
      <div className="control">
        <div className="checkbox-round">
          <input id={this.props.id} type="checkbox" />
          <label htmlFor={this.props.id} />
        </div>

        <h2>{this.props.name}</h2>
      </div>
    );
  }
}

export default SelectButton;
