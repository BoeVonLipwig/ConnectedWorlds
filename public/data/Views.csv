view,filename
projectView,project
schoolView,school
collaborators,nodes
