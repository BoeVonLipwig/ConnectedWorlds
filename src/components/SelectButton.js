import React from "react";

class SelectButton extends React.Component {
  render() {
    return (
      <div className="control">
        <div className="checkbox-round">
          <input
            id={this.props.id}
            type="checkbox"
            checked={this.props.isChecked}
          />
          <label htmlFor={this.props.id} onClick={this.props.clickHandler} />
        </div>
        <h2>{this.props.name}</h2>
      </div>
    );
  }
}

export default SelectButton;
