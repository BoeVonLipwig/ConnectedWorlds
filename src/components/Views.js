import React, { Fragment } from "react";
import RadioButton from "./RadioButton";
import cytoscapeStore from "../util/CytoscapeStore";
import layoutFactory from "../util/LayoutFactory";
import { observer } from "mobx-react";
import DropDownMenu from "./DropDownMenu";

class Views extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hasTypes: false,
      views: [
        {
          name: "Semi-Circle",
          id: "showSegment",
          isChecked: true
        },
        {
          name: "Many-Circle",
          id: "showCircles",
          isChecked: false
        },
        {
          name: "Collaborators",
          id: "showCollab",
          isChecked: false
        }
      ]
    };
    this.clickHandler = this.clickHandler.bind(this);
  }

  toggleCheck(id) {
    const newViews = this.state.views.map(function(entry) {
      if (entry.id === id) {
        cytoscapeStore.layouts = layoutFactory.computeLayout(entry.id);
        return Object.assign({}, entry, {
          isChecked: true
        });
      } else {
        return Object.assign({}, entry, {
          isChecked: false
        });
      }
    });
    this.setState(
      Object.assign({}, this.state, {
        views: newViews
      })
    );
  }

  clickHandler(e, id) {
    this.toggleCheck(id);
  }

  render() {
    let colabView = this.state.views[this.state.views.length - 1];
    return (
      <Fragment>
        {/*{this.createRadio()}*/}
        {this.createMenu(cytoscapeStore.specialTypes.$mobx.values)}
        {/*<SelectButton*/}
        {/*key={colabView.id}*/}
        {/*name={colabView.name}*/}
        {/*id={colabView.id}*/}
        {/*isChecked={colabView.isChecked}*/}
        {/*clickHandler={event => this.clickHandler(event, colabView.id)}*/}
        {/*/>*/}
      </Fragment>
    );
  }

  createRadio() {
    return this.state.views.map(elem => {
      return elem.name !== "Collaborators" ? (
        <RadioButton
          key={elem.id}
          name={elem.name}
          id={elem.id}
          isChecked={elem.isChecked}
          clickHandler={event => this.clickHandler(event, elem.id)}
        />
      ) : null;
    });
  }

  createMenu(cyArray) {
    console.log(cyArray);
    return cyArray.length > 0 ? <DropDownMenu data={cyArray} /> : null;
  }
}

export default observer(Views);
