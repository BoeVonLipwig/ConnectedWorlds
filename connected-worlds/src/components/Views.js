import React from "react";
import SelectButton from "./SelectButton";
import { notify } from "../util/EventManager";

class Views extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      views: [
        { name: "Projects", id: "showProjects", isChecked: true },
        { name: "Programme", id: "showSchools", isChecked: false },
        { name: "Collaborators", id: "showCollab", isChecked: false }
      ]
    };
    this.clickHandler = this.clickHandler.bind(this);
  }

  toggleCheck(id) {
    const newViews = this.state.views.map(function(entry) {
      if (entry.id === id) {
        return Object.assign({}, entry, {
          isChecked: true
        });
      } else {
        return Object.assign({}, entry, {
          isChecked: false
        });
      }
    });
    this.setState(
      Object.assign({}, this.state, {
        views: newViews
      })
    );
  }

  clickHandler(e, id) {
    this.toggleCheck(id);
  }

  render() {
    const items = this.state.views.map(elem => {
      return (
        <SelectButton
          key={elem.id}
          name={elem.name}
          id={elem.id}
          isChecked={elem.isChecked}
          clickHandler={event => this.clickHandler(event, elem.id)}
        />
      );
    });
    return items;
  }
}

export default Views;
