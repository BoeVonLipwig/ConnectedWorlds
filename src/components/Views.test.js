import React from "react";
import ReactDOM from "react-dom";
import Views from "./Views";
import SelectButton from "./SelectButton";
import { mount, find, at, simulate, text, instance } from "enzyme";

it("renders views without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Views />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders views correctly without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Views />, div);
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});

it("should unselect the projects and collaborators buttons when the programmes button is clicked", () => {
  const wrapper = mount(
    <div>
      <Views test="true" />
    </div>
  );

  wrapper
    .find(Views)
    .instance()
    .setState({
      views: [
        {
          name: "1",
          id: "1",
          isChecked: true
        },
        {
          name: "2",
          id: "2",
          isChecked: false
        },
        {
          name: "3",
          id: "3",
          isChecked: false
        }
      ]
    });

  wrapper.update();
  const projectsButtonOld = wrapper.find(SelectButton).at(0);
  const programmesButtonOld = wrapper.find(SelectButton).at(1);
  const collaboratorsButtonOld = wrapper.find(SelectButton).at(2);

  expect(projectsButtonOld.find("input").props().checked).toEqual(true);
  expect(programmesButtonOld.find("input").props().checked).toEqual(false);
  expect(collaboratorsButtonOld.find("input").props().checked).toEqual(false);

  programmesButtonOld.prop("clickHandler")();

  wrapper.update();
  const projectsButtonNew = wrapper.find(SelectButton).at(0);
  const programmesButtonNew = wrapper.find(SelectButton).at(1);
  const collaboratorsButtonNew = wrapper.find(SelectButton).at(2);

  expect(projectsButtonNew.find("input").props().checked).toEqual(false);
  expect(programmesButtonNew.find("input").props().checked).toEqual(true);
  expect(collaboratorsButtonNew.find("input").props().checked).toEqual(false);
});
