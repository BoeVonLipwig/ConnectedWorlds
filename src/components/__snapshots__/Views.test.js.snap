// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders views correctly without crashing 1`] = `
<div>
  <div
    class="control"
  >
    <div
      class="checkbox-round"
    >
      <input
        id="showProjects"
        type="checkbox"
        readonly=""
      />
      <label
        for="showProjects"
      />
    </div>
    <h2>
      Projects
    </h2>
  </div>
  <div
    class="control"
  >
    <div
      class="checkbox-round"
    >
      <input
        id="showSchools"
        type="checkbox"
        readonly=""
      />
      <label
        for="showSchools"
      />
    </div>
    <h2>
      Programme
    </h2>
  </div>
  <div
    class="control"
  >
    <div
      class="checkbox-round"
    >
      <input
        id="showCollab"
        type="checkbox"
        readonly=""
      />
      <label
        for="showCollab"
      />
    </div>
    <h2>
      Collaborators
    </h2>
  </div>
</div>
`;
